sqlglot/expressions.py

```diff
@@ -3528,23 +3532,54 @@ class Type(AutoName):
         Type.DATETIME64,
     }
 
-    META_TYPES = {"UNKNOWN", "NULL"}
+    NESTED_TYPES = {
+        Type.ARRAY,
+        Type.LOWCARDINALITY,
+        Type.MAP,
+        Type.NESTED,
+        Type.STRUCT,
+    }
+
+    META_TYPES = {
+        "UNKNOWN",
+        "NULL",
+    }
 
     @classmethod
     def build(
-        cls, dtype: str | DataType | DataType.Type, dialect: DialectType = None, **kwargs
+        cls,
+        dtype: str | DataType | DataType.Type,
+        dialect: DialectType = None,
+        udt: bool = False,
+        **kwargs,
     ) -> DataType:
+        """
+        Constructs a DataType object.
+
+        Args:
+            dtype: the data type of interest.
+            dialect: the dialect to use for parsing `dtype`, in case it's a string.
+            udt: when set to True, `dtype` will be used as-is if it can't be parsed into a
+                DataType, thus creating a user-defined type.
+            kawrgs: additional arguments to pass in the constructor of DataType.
+
+        Returns:
+            The constructed DataType object.
+        """
         from sqlglot import parse_one
 
         if isinstance(dtype, str):
             upper = dtype.upper()
             if upper in DataType.META_TYPES:
-                data_type_exp: t.Optional[Expression] = DataType(this=DataType.Type[upper])
+                data_type_exp = DataType(this=DataType.Type[upper])
             else:
-                data_type_exp = parse_one(dtype, read=dialect, into=DataType)
+                try:
+                    data_type_exp = parse_one(dtype, read=dialect, into=DataType)
+                except ParseError as e:
+                    if udt:
+                        return DataType(this=DataType.Type.USERDEFINED, expression=dtype, **kwargs)
 
-            if data_type_exp is None:
-                raise ValueError(f"Unparsable data type value: {dtype}")
```

Comment:
I couldn't trace a path in the code where `data_type_exp` could be `None` here - if parsing fails, I believe we'll always get an exception here.